crates/viewer/re_space_view_dataframe/src/dataframe_ui.rs

```diff
@@ -0,0 +1,350 @@
+            .inner_margin(egui::Margin::symmetric(4.0, 0.0))
+            .show(ui, |ui| {
+                if cell.row_nr == 0 {
+                    if let Some(entity_path) = &self.header_entity_paths[cell.group_index] {
+                        ui.label(entity_path.to_string());
+                    }
+                } else if cell.row_nr == 1 {
+                    ui.strong(self.schema[cell.col_range.start].short_name());
+                } else {
+                    // this should never happen
+                    error_ui(ui, format!("Unexpected header row_nr: {}", cell.row_nr));
+                }
+            });
+    }
+
+    fn cell_ui(&mut self, ui: &mut egui::Ui, cell: &egui_table::CellInfo) {
+        re_tracing::profile_function!();
+
+        if cell.row_nr % 2 == 1 {
+            // Paint stripes
+            ui.painter()
+                .rect_filled(ui.max_rect(), 0.0, ui.visuals().faint_bg_color);
+        }
+
+        // sanity check, this should never happen
+        if cell.row_nr >= self.num_rows {
+            error_ui(
+                ui,
+                format!(
+                    "Unexpected row_nr: {} (table row count {})",
+                    cell.row_nr, self.num_rows
+                ),
+            );
+            return;
+        }
+
+        let display_data = match &self.display_data {
+            Ok(display_data) => display_data,
+            Err(err) => {
+                error_ui(ui, format!("Error with display data: {err}"));
+                return;
+            }
+        };
+
+        egui::Frame::none()
+            .inner_margin(egui::Margin::symmetric(4.0, 0.0))
+            .show(ui, |ui| {
+                if let Some((batch_nr, batch_index)) = display_data
+                    .batch_and_row_from_row_nr
+                    .get(&cell.row_nr)
+                    .copied()
+                {
+                    let batch = &display_data.display_record_batches[batch_nr];
+                    let column = &batch.columns()[cell.col_nr];
+
+                    // compute the latest at query for this row (used to display tooltips)
+                    let timestamp = display_data
+                        .time_column_index
+                        .and_then(|index| {
+                            display_data.display_record_batches[batch_nr].columns()[index]
+                                .try_decode_time(batch_index)
+                        })
+                        .unwrap_or(TimeInt::MAX);
+                    let latest_at_query =
+                        LatestAtQuery::new(self.query_handle.timeline(), timestamp);
+                    let row_id = display_data
+                        .row_id_column_index
+                        .and_then(|index| {
+                            display_data.display_record_batches[batch_nr].columns()[index]
+                                .try_decode_row_id(batch_index)
+                        })
+                        .unwrap_or(RowId::ZERO);
```

Comment:
To reduce rightward drift, we could call into another function from inside the `Frame`